lib/auth/grpcserver.go

```diff
@@ -1065,6 +1065,81 @@ func (g *GRPCServer) DeleteAllKubeServices(ctx context.Context, req *proto.Delet
 	return &empty.Empty{}, nil
 }
 
+// GetRole retrieves a role by name.
+func (g *GRPCServer) GetRole(ctx context.Context, req *proto.GetRoleRequest) (*types.RoleV3, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	role, err := auth.GetRole(req.Name)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	roleV3, ok := role.(*types.RoleV3)
```

Comment:
I'm confused about this pattern of wrapping the types in an interface but with both parties expecting a specific type. I know this sort of makes sense to the client, but internally it makes no sense - if gRPC server accepts values of V3 type, then rest of the internals should be handling the same type. In case, the types are promoted to V4, every single place will need to be modified anyways.

It is not exactly an issue with this PR - just something I'm maybe looking for additional views on.